egs/hub4_english/s5/local/tuning/run_segmentation_wsj_f.sh

```diff
@@ -0,0 +1,415 @@
+
+steps/align_si.sh --nj 40 --cmd "$train_cmd" \
+  data/train_reseg_${affix} \
+  data/lang_nosp exp/tri1${affix} exp/tri1${affix}_ali_reseg_${affix}
+
+steps/train_lda_mllt.sh --cmd "$train_cmd" 3500 25000 \
+  data/train_reseg_${affix} data/lang_nosp exp/tri2${affix}
+
+affix=d
+steps/cleanup/segment_long_utterances.sh --cmd "$train_cmd" \
+  --stage $segment_stage \
+  --config conf/segment_long_utts.conf \
+  --max-segment-duration 30 --overlap-duration 5 \
+  --num-neighbors-to-search 1 --nj 80 \
+  exp/tri2a data/lang_nosp data/train_long data/train_reseg_${affix} \
+  exp/segment_long_utts_${affix}_train
+
+steps/compute_cmvn_stats.sh data/train_reseg_${affix} \
+  exp/make_mfcc/train_reseg_${affix} mfcc
+utils/fix_data_dir.sh data/train_reseg_${affix}
+
+###############################################################################
+# Train new model on segmented data directory starting from the same model
+# used for segmentation. (tri2b)
+###############################################################################
+
+steps/align_si.sh --nj 40 --cmd "$train_cmd" \
+  data/train_reseg_${affix} \
+  data/lang_nosp exp/tri2b exp/tri2b_ali_reseg_${affix}  || exit 1;
+
+# Train SAT system on reseg data
+steps/train_sat.sh --cmd "$train_cmd" 4200 40000 \
+  data/train_reseg_${affix} data/lang_nosp \
+  exp/tri2b_ali_reseg_${affix} exp/tri3${affix}
+
+(
+utils/mkgraph.sh data/lang_nosp_test_tgpr \
+  exp/tri3${affix} exp/tri3${affix}/graph_nosp || exit 1;
+for dset in eval98.pem eval97.pem eval99_1.pem eval99_2.pem; do
+  this_nj=`cat data/$dset/spk2utt | wc -l`
+  if [ $this_nj -gt 20 ]; then
+    this_nj=20
+  fi
+  steps/decode_fmllr.sh --nj $this_nj --cmd "$decode_cmd" --num-threads 4 \
+    exp/tri3${affix}/graph_nosp data/$dset exp/tri3${affix}/decode_nosp_${dset}
+  steps/lmrescore_const_arpa.sh --cmd "$decode_cmd" \
+    data/lang_nosp_test data/lang_nosp_test_rescore \
+    data/${dset} exp/tri3${affix}/decode_nosp_${dset} \
+    exp/tri3${affix}/decode_nosp_${dset}_rescore
+done
+) &
+
+exit 0
+
+steps/align_fmllr.sh --nj $nj --cmd "$train_cmd" \
+  data/train data/lang_nosp exp/tri3${affix} exp/tri3${affix}_ali
+
+steps/train_sat.sh --cmd "$train_cmd" 5000 100000 \
+  data/train data/lang_nosp exp/tri3${affix}_ali exp/tri4${affix}
+
+(
+utils/mkgraph.sh data/lang_nosp_test exp/tri4${affix} \
+  exp/tri4${affix}/graph_nosp
+for dset in eval98.pem eval97.pem eval99_1.pem eval99_2.pem; do
+  this_nj=`cat data/$dset/spk2utt | wc -l`
+  if [ $this_nj -gt 20 ]; then
+    this_nj=20
+  fi
+  steps/decode_fmllr.sh --nj $this_nj --cmd "$decode_cmd" --num-threads 4 \
+    exp/tri4${affix}/graph_nosp data/$dset exp/tri4${affix}/decode_nosp_${dset}
+  steps/lmrescore_const_arpa.sh --cmd "$decode_cmd" \
+    data/lang_nosp_test data/lang_nosp_test_rescore \
+    data/${dset} exp/tri4${affix}/decode_nosp_${dset} \
+    exp/tri4${affix}/decode_nosp_${dset}_rescore
+done
+) &
+
+exit 0
```

Comment:
is this 'exit 0' supposed to be here?
I don't understand what these scripts are for and how they fit in the workflow.

Reply:
shouldn't  there be `wait`?